File: java/src/main/java/br/com/vps/consulting/b2b/management/partner/domain/PartnerRepository.java
package br.com.vps.consulting.b2b.management.partner.domain;

import br.com.vps.consulting.b2b.management.shared.core.page.PageCustom;

public interface PartnerRepository {
    Partner save(Partner partner);
    PageCustom<Partner> findAll(long pageSize, long pageNumber);
}

File: kotlin/src/main/kotlin/br/com/vps/consulting/b2b/management/partner/domain/PartnerRepository.kt
package br.com.vps.consulting.b2b.management.partner.domain

import br.com.vps.consulting.b2b.management.shared.core.page.PageCustom

interface PartnerRepository {
    fun save(partner: Partner): Partner
    fun findById(id: PartnerId): Partner?
    fun findAll(document: String?, pageSize: Int, pageNumber: Int): PageCustom<Partner>
}
